batch/batch/server/server.py

```diff
@@ -130,13 +142,17 @@ def __init__(self, pod_spec, batch_id, attributes, callback):
                                               }),
             spec=pod_spec)
 
-        self._pod_name = None
-        self.exit_code = None
-
-        self._state = 'Created'
         log.info('created job {}'.format(self.id))
 
-        self._create_pod()
+        if not self.parents:
+            self._create_pod()
+        else:
+            self.refresh_parents_and_maybe_create()
+
+    def refresh_parents_and_maybe_create(self):
```

Comment:
It doesn't look like this does maybe create anymore?

Reply:
`parent_new_state` creates a pod when no parents are incomplete